File: backend/src/main/java/com/teaching/employment/entity/School.java
package com.teaching.employment.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 学校实体类
 *
 * @author Teaching Employment Platform Team
 * @since 2026-01-12
 */
@Data
@TableName("t_school")
@ApiModel(description = "学校信息")
public class School implements Serializable {

    private static final long serialVersionUID = 1L;

    @ApiModelProperty(value = "学校ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @ApiModelProperty(value = "学校名称")
    private String schoolName;

    @ApiModelProperty(value = "学校代码")
    private String schoolCode;

    @ApiModelProperty(value = "办学许可证号")
    private String licenseNumber;

    @ApiModelProperty(value = "许可证颁发日期")
    private LocalDate licenseIssueDate;

    @ApiModelProperty(value = "许可证有效期至")
    private LocalDate licenseExpiryDate;

    @ApiModelProperty(value = "许可证图片URL")
    private String licenseImageUrl;

    @ApiModelProperty(value = "省份")
    private String province;

    @ApiModelProperty(value = "城市")
    private String city;

    @ApiModelProperty(value = "详细地址")
    private String address;

    @ApiModelProperty(value = "学校官网")
    private String website;

    @ApiModelProperty(value = "联系人")
    private String contactPerson;

    @ApiModelProperty(value = "联系电话")
    private String contactPhone;

    @ApiModelProperty(value = "邮箱")
    private String email;

    @ApiModelProperty(value = "学校简介")
    private String description;

    @ApiModelProperty(value = "状态：1-正常 0-停用")
    private Integer status;

    @ApiModelProperty(value = "是否删除：1-已删除 0-未删除")
    @TableLogic
    private Integer isDeleted;

    @ApiModelProperty(value = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @ApiModelProperty(value = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/school.ts
import { http } from '@/utils/request'
import type { IPage } from './types'

// 学校接口类型定义
export interface School {
  id?: number
  schoolName?: string
  schoolCode?: string
  licenseNumber?: string
  licenseIssueDate?: string
  licenseExpiryDate?: string
  licenseImageUrl?: string
  province?: string
  city?: string
  address?: string
  website?: string
  contactPerson?: string
  contactPhone?: string
  email?: string
  description?: string
  status?: number
  createTime?: string
  updateTime?: string
}

// 分页查询参数
export interface SchoolPageParams {
  current: number
  size: number
  schoolName?: string
  province?: string
  city?: string
}

/**
 * 获取学校分页列表
 */
export function getSchoolPageApi(params: SchoolPageParams) {
  return http.get<IPage<School>>('/school/page', { params })
}

/**
 * 获取所有学校列表
 */
export function getSchoolListApi() {
  return http.get<School[]>('/school/list')
}

/**
 * 根据ID获取学校详情
 */
export function getSchoolByIdApi(id: number) {
  return http.get<School>(`/school/${id}`)
}

/**
 * 创建学校
 */
export function createSchoolApi(data: School) {
  return http.post<void>('/school', data)
}

/**
 * 更新学校
 */
export function updateSchoolApi(data: School) {
  return http.put<void>('/school', data)
}

/**
 * 删除学校
 */
export function deleteSchoolApi(id: number) {
  return http.delete<void>(`/school/${id}`)
}
